### PR TITLE
fix: organize routes

### PR DESCRIPTION
**Pull request by ajpauwels**
This PR organizes the routes into modules and submodules so we don't have all of our route logic bundled up into one file.

It also re-works the session filters so that they can be called from the context of warp's `.with(...)` filter function.

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ajpauwels**
You can assign the PR to them by writing `/assign @ajpauwels` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/routes/proxy.rs [hunk 1]
```diff
@@ -29,13 +27,19 @@ pub fn post<Q: Relayer>(
...
+                );
                 if dest_root != origin_root {
+                    println!("not relaying");
                     Err(warp::reject::custom(RelayRejection))
                 } else {
-                    relayer
-                        .get(body_params.host_url)
-                        .await
-                        .map_err(|_| warp::reject::custom(RelayRejection))
+                    println!("relaying");
```
Comment (howhelan) [inline src/routes/proxy.rs:38]: remove

File: src/routes/proxy.rs [hunk 2]
```diff
@@ -29,13 +27,19 @@ pub fn post<Q: Relayer>(
                 let dest_root = parse_url_root(&body_params.host_url)
                     .map_err(|_| warp::reject::custom(RelayRejection))?;
 
+                println!(
+                    "{:?} =/= {:?}, {}",
+                    origin_root, dest_root, &body_params.host_url
+                );
                 if dest_root != origin_root {
+                    println!("not relaying");
```
Comment (howhelan) [inline src/routes/proxy.rs:35]: remove

File: src/routes/proxy.rs [hunk 3]
```diff
@@ -29,13 +27,19 @@ pub fn post<Q: Relayer>(
                 let dest_root = parse_url_root(&body_params.host_url)
                     .map_err(|_| warp::reject::custom(RelayRejection))?;
 
+                println!(
```
Comment (howhelan) [inline src/routes/proxy.rs:30]: remove

File: src/routes/secure/data.rs [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+                let reply = post::reply(data, &path, &new_token, query, &render_engine)?;
+                Ok::<_, Rejection>((
+                    Box::new(reply) as Box<dyn Reply>,
+                    format!("/secure/data/{}/{}", &path, &old_token),
+                    Some(format!("/secure/data/{}/{}", &path, &new_token)),
+                    Some(new_token),
+                ))
+
+                //     match session_with_store.session.get("token") {
```
Comment (howhelan) [inline src/routes/secure/data.rs:209]: remove

File: src/routes/secure/data.rs [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+                //             }
+                //         }
+                //         None => Err(warp::reject::custom(SessionTokenNotFoundRejection)),
+                //     }
+            },
+        )
+        .untuple_one()
+    // .and_then(build_session)
+    // .untuple_one()
+    // .and_then(warp_sessions::reply::with_session)
```
Comment (howhelan) [inline src/routes/secure/data.rs:247]: remove

File: src/routes/proxy.rs [hunk 6]
```diff
@@ -29,13 +27,19 @@ pub fn post<Q: Relayer>(
...
+                    println!("not relaying");
                     Err(warp::reject::custom(RelayRejection))
                 } else {
-                    relayer
-                        .get(body_params.host_url)
-                        .await
-                        .map_err(|_| warp::reject::custom(RelayRejection))
+                    println!("relaying");
+                    relayer.get(body_params.host_url).await.map_err(|e| {
+                        println!("{:?}", e);
```
Comment (howhelan) [inline src/routes/proxy.rs:40]: remove

**Review by howhelan — COMMENTED** [review]
Added comments to remove all println's, we should add something like this to client. https://github.com/pauwels-labs/redact-store/commit/00603b21c62bc6da4e47cdda55330d2a715348de

Feel free to merge and I can add better logging as a separate PR.

File: src/error_handler.rs [hunk 7]
```diff
@@ -20,6 +20,8 @@ pub async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible>
     let code;
     let message;
 
+    println!("{:?}", err);
```
[the remark this answers is not in the document]
Comment (ajpauwels) [reply src/error_handler.rs:23]: 100% agree, this is a hack

File: src/relayer.rs [hunk 8]
```diff
@@ -43,21 +43,27 @@ pub struct MutualTLSRelayer {
...
 
-        let client = reqwest::Client::builder()
-            .identity(pkcs12)
-            .use_rustls_tls()
-            .build()
-            .unwrap();
-
-        Ok(MutualTLSRelayer { client })
+        let mut client_builder = reqwest::Client::builder().identity(pkcs12).use_rustls_tls();
+        if let Some(roots) = additional_roots {
```
[the remark this answers is not in the document]
Comment (ajpauwels) [reply src/relayer.rs:58]: This is to support a storage that uses a non-public CA signed TLS cert, e.g. when we make our own certs. It allows passing in an array of additional CA certs. With reqwest, in order to have these certs be used for the TLS trust anchor, you have to disable the built-in CA certs (i.e. the public ones).

Added comments to the code

File: src/routes/proxy.rs (hunk 1 above)
Comment (ajpauwels) [reply src/routes/proxy.rs:38]: removed

File: src/routes/proxy.rs (hunk 2 above)
Comment (ajpauwels) [reply src/routes/proxy.rs:35]: removed

File: src/routes/proxy.rs (hunk 3 above)
Comment (ajpauwels) [reply src/routes/proxy.rs:30]: removed

File: src/routes/secure/data.rs (hunk 4 above)
Comment (ajpauwels) [reply src/routes/secure/data.rs:209]: removed

File: src/routes/secure/data.rs (hunk 5 above)
Comment (ajpauwels) [reply src/routes/secure/data.rs:247]: removed

**Pull request merged**